polyhost/gui/layout_dialog/keycode_composer.py

```
@@ -0,0 +1,246 @@
+    ("MT", "MT — Mod-tap (hold mod / tap key)", False, True, True),
+    ("MOD", "Modified key (Ctrl/Shift/… + key)", False, True, True),
+    ("SH_T", "SH_T — Swap-hands tap-hold (tap key / hold swap)", False, False, True),
+]
+
+
+class KeycodeComposer(QWidget):
+    keycodeSelected = pyqtSignal(str, str, int, int)  # nice_name, name, keycode, font_hint
+
+    def __init__(self, basic_keycodes: dict[str, int], num_layers: int = 9):
+        super().__init__()
+        # Inner-key choices: basic keycodes only (0x00–0xFF) — that is all MT/LT
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)